### PR TITLE
Functional Tests : Audit - BO - International

### PR DESCRIPTION
**Pull request by florine2623**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 9.0.x
| Description?      | Functional Tests : Audit - BO - International
| Type?             | improvement
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | CI is :green_circle: 
| UI Tests          | :dolphin: :  https://github.com/florine2623/testing_pr/actions/runs/12356481044 ✅ <br/>:seal: : https://github.com/florine2623/testing_pr/actions/runs/12356493937 ✅ 
| Fixed issue or discussion?     | N/A
| Related PRs       | N/A
| Sponsor company   | @PrestaShopCorp

### CODE REVIEW

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+    await boLocalizationPage.goToSubTabLanguages(page);
+
+    const pageTitle = await boLanguagesPage.getPageTitle(page);
+    expect(pageTitle).to.contains(boLanguagesPage.pageTitle);
+  });
+
+  it('should go to edit language page', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'goToEditLanguagePage', baseContext);
+
+    await boLanguagesPage.goToEditLanguage(page, 1);
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
    await boLocalizationPage.goToSubTabLanguages(page);
    await boLanguagesPage.goToEditLanguage(page, 1);
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+    expect(pageTitle).to.contains(boCurrenciesPage.pageTitle);
+
+    const jsErrors = utilsPlaywright.getJsErrors();
+    expect(jsErrors.length).to.equals(0);
+  });
+
+  it('should go to edit currency page', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'goToEditCurrency', baseContext);
+
+    await boCurrenciesPage.goToEditCurrencyPage(page, 1);
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
    await boLocalizationPage.goToSubTabCurrencies(page);
    await boCurrenciesPage.goToEditCurrencyPage(page, 1);
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+    );
+    await boLocalizationPage.closeSfToolBar(page);
+
+    const pageTitle = await zonesPage.getPageTitle(page);
+    expect(pageTitle).to.contains(zonesPage.pageTitle);
+  });
+
+  it('should go to edit zone page', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'goToEditZonePage', baseContext);
+
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
    await addZonePage.goToSubMenu(
      page,
      boDashboardPage.internationalParentLink,
      boDashboardPage.locationsLink,
    );
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 4]
```diff
@@ -0,0 +1,433 @@
...
+    await zonesPage.goToSubTabCountries(page);
+
+    const pageTitle = await boCountriesPage.getPageTitle(page);
+    expect(pageTitle).to.contains(boCountriesPage.pageTitle);
+  });
+
+  it('should go to edit country page', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'goToEditCountryPage', baseContext);
+
+    await boCountriesPage.goToEditCountryPage(page, 1);
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
    await zonesPage.goToSubTabCountries(page);
    await boCountriesPage.goToEditCountryPage(page, 1);
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 5]
```diff
@@ -0,0 +1,433 @@
...
+    await zonesPage.goToSubTabStates(page);
+
+    const pageTitle = await statesPage.getPageTitle(page);
+    expect(pageTitle).to.contains(statesPage.pageTitle);
+  });
+
+  it('should go to edit state page', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'goToEditStatePage', baseContext);
+
+    await statesPage.goToEditStatePage(page, 1);
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
    await zonesPage.goToSubTabStates(page);
    await statesPage.goToEditStatePage(page, 1);
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 6]
```diff
@@ -0,0 +1,433 @@
...
+    );
+
+    const pageTitle = await taxesPage.getPageTitle(page);
+    expect(pageTitle).to.contains(taxesPage.pageTitle);
+  });
+
+  it('should go to edit tax page', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'goToEditPage', baseContext);
+
+    await taxesPage.goToEditTaxPage(page, 1);
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
    await addTaxPage.goToSubMenu(
      page,
      boDashboardPage.internationalParentLink,
      boDashboardPage.taxesLink,
    );
    await taxesPage.goToEditTaxPage(page, 1);
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 7]
```diff
@@ -0,0 +1,433 @@
...
+    await taxesPage.goToTaxRulesPage(page);
+
+    const pageTitle = await taxRulesPage.getPageTitle(page);
+    expect(pageTitle).to.contains(taxRulesPage.pageTitle);
+  });
+
+  it('should go to edit tax Rule page', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'goToEditPage', baseContext);
+
+    await taxRulesPage.goToEditTaxRulePage(page, 1);
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
    await taxesPage.goToTaxRulesPage(page);
    await taxRulesPage.goToEditTaxRulePage(page, 1);
```

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 8]
```diff
@@ -0,0 +1,374 @@
...
+    await boLoginPage.successLogin(page, global.BO.EMAIL, global.BO.PASSWD);
+
+    const pageTitle = await boDashboardPage.getPageTitle(page);
+    expect(pageTitle).to.contains(boDashboardPage.pageTitle);
+
+    const jsErrors = utilsPlaywright.getJsErrors();
+    expect(jsErrors.length).to.equals(0);
+  });
+
+  it('should go to \'Localization\' page > Localization tab', async function () {
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
  it('should go to \'Localization > Localization\' page', async function () {
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 9]
```diff
@@ -0,0 +1,374 @@
...
+    await boLocalizationPage.closeSfToolBar(page);
+
+    const pageTitle = await boLocalizationPage.getPageTitle(page);
+    expect(pageTitle).to.contains(boLocalizationPage.pageTitle);
+
+    const jsErrors = utilsPlaywright.getJsErrors();
+    expect(jsErrors.length).to.equals(0);
+  });
+
+  it('should go to \'Languages\' page', async function () {
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
  it('should go to \'Localization > Languages\' page', async function () {
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 10]
```diff
@@ -0,0 +1,374 @@
...
+    await boLocalizationPage.goToSubTabLanguages(page);
+
+    const pageTitle = await boLanguagesPage.getPageTitle(page);
+    expect(pageTitle).to.contains(boLanguagesPage.pageTitle);
+
+    const jsErrors = utilsPlaywright.getJsErrors();
+    expect(jsErrors.length).to.equals(0);
+  });
+
+  it('should go to add new language page', async function () {
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
  it('should go to \'Localization > Languages > New Language\' page', async function () {
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 11]
```diff
@@ -0,0 +1,374 @@
...
+    await boLanguagesPage.goToAddNewLanguage(page);
+
+    const pageTitle = await boLanguagesCreatePage.getPageTitle(page);
+    expect(pageTitle).to.contains(boLanguagesCreatePage.pageTitle);
+
+    const jsErrors = utilsPlaywright.getJsErrors();
+    expect(jsErrors.length).to.equals(0);
+  });
+
+  it('should go to edit Language page', async function () {
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]: ```suggestion```suggestion
  it('should go to \'Localization > Languages > Edit Language\' page', async function () {
```
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 12]
```diff
@@ -0,0 +1,374 @@
...
+    await boLanguagesPage.goToEditLanguage(page, 1);
+
+    const pageTitle = await boLanguagesCreatePage.getPageTitle(page);
+    expect(pageTitle).to.contains(boLanguagesCreatePage.pageEditTitle);
+
+    const jsErrors = utilsPlaywright.getJsErrors();
+    expect(jsErrors.length).to.equals(0);
+  });
+
+  it('should go to \'Currencies\' page', async function () {
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
  it('should go to \'Localization > Currencies\' page', async function () {
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 13]
```diff
@@ -0,0 +1,374 @@
...
+    await boLocalizationPage.goToSubTabCurrencies(page);
+
+    const pageTitle = await boCurrenciesPage.getPageTitle(page);
+    expect(pageTitle).to.contains(boCurrenciesPage.pageTitle);
+
+    const jsErrors = utilsPlaywright.getJsErrors();
+    expect(jsErrors.length).to.equals(0);
+  });
+
+  it('should go to create new currency page', async function () {
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
  it('should go to \'Localization > Currencies > New Currency\' page', async function () {
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 14]
```diff
@@ -0,0 +1,374 @@
...
+    await boCurrenciesPage.goToAddNewCurrencyPage(page);
+
+    const pageTitle = await boCurrenciesCreatePage.getPageTitle(page);
+    expect(pageTitle).to.contains(boCurrenciesCreatePage.pageTitle);
+
+    const jsErrors = utilsPlaywright.getJsErrors();
+    expect(jsErrors.length).to.equals(0);
+  });
+
+  it('should go to edit Currencies page', async function () {
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
  it('should go to \'Localization > Currencies > Edit Currency\' page', async function () {
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 15]
```diff
@@ -0,0 +1,374 @@
...
+    await boCurrenciesPage.goToEditCurrencyPage(page, 1);
+
+    const pageTitle = await boCurrenciesCreatePage.getPageTitle(page);
+    expect(pageTitle).to.contains(boCurrenciesCreatePage.editCurrencyPage);
+
+    const jsErrors = utilsPlaywright.getJsErrors();
+    expect(jsErrors.length).to.equals(0);
+  });
+
+  it('should go to \'Geolocation\' page', async function () {
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
  it('should go to \'Localization > Geolocation\' page', async function () {
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 16]
```diff
@@ -0,0 +1,374 @@
...
+    await testContext.addContextItem(this, 'testIdentifier', 'goToGeolocationPage', baseContext);
+
+    await boLocalizationPage.goToSubTabGeolocation(page);
+
+    const pageTitle = await geolocationPage.getPageTitle(page);
+    expect(pageTitle).to.equal(geolocationPage.pageTitle);
+  });
+
+  it('should go to \'Locations\' page > Zones tab', async function () {
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
  it('should go to \'Locations > Zones\' page', async function () {
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 17]
```diff
@@ -0,0 +1,374 @@
...
+    await boLocalizationPage.closeSfToolBar(page);
+
+    const pageTitle = await zonesPage.getPageTitle(page);
+    expect(pageTitle).to.contains(zonesPage.pageTitle);
+
+    const jsErrors = utilsPlaywright.getJsErrors();
+    expect(jsErrors.length).to.equals(0);
+  });
+
+  it('should go to add new zone page', async function () {
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
  it('should go to \'Locations > Zones > New Zone\' page', async function () {
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 18]
```diff
@@ -0,0 +1,374 @@
...
+    await zonesPage.goToAddNewZonePage(page);
+
+    const pageTitle = await addZonePage.getPageTitle(page);
+    expect(pageTitle).to.contains(addZonePage.pageTitleCreate);
+
+    const jsErrors = utilsPlaywright.getJsErrors();
+    expect(jsErrors.length).to.equals(0);
+  });
+
+  it('should go to edit Zones page', async function () {
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
  it('should go to \'Locations > Zones > Edit Zone\' page', async function () {
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 19]
```diff
@@ -0,0 +1,374 @@
...
+    await zonesPage.goToEditZonePage(page, 1);
+
+    const pageTitle = await addZonePage.getPageTitle(page);
+    expect(pageTitle).to.contains(addZonePage.pageTitleEdit);
+
+    const jsErrors = utilsPlaywright.getJsErrors();
+    expect(jsErrors.length).to.equals(0);
+  });
+
+  it('should go to \'Countries\' page', async function () {
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
  it('should go to \'Locations > Countries\' page', async function () {
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 20]
```diff
@@ -0,0 +1,374 @@
...
+    await zonesPage.goToSubTabCountries(page);
+
+    const pageTitle = await boCountriesPage.getPageTitle(page);
+    expect(pageTitle).to.contains(boCountriesPage.pageTitle);
+
+    const jsErrors = utilsPlaywright.getJsErrors();
+    expect(jsErrors.length).to.equals(0);
+  });
+
+  it('should go to add new country page', async function () {
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
  it('should go to \'Locations > Countries > New Country\' page', async function () {
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 21]
```diff
@@ -0,0 +1,374 @@
...
+    await boCountriesPage.goToAddNewCountryPage(page);
+
+    const pageTitle = await boCountriesCreatePage.getPageTitle(page);
+    expect(pageTitle).to.contains(boCountriesCreatePage.pageTitleCreate);
+
+    const jsErrors = utilsPlaywright.getJsErrors();
+    expect(jsErrors.length).to.equals(0);
+  });
+
+  it('should go to edit Countries page', async function () {
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
  it('should go to \'Locations > Countries > Edit Country\' page', async function () {
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 22]
```diff
@@ -0,0 +1,374 @@
...
+    await boCountriesPage.goToEditCountryPage(page, 1);
+
+    const pageTitle = await boCountriesCreatePage.getPageTitle(page);
+    expect(pageTitle).to.contains(boCountriesCreatePage.pageTitleEdit);
+
+    const jsErrors = utilsPlaywright.getJsErrors();
+    expect(jsErrors.length).to.equals(0);
+  });
+
+  it('should go to \'States\' page', async function () {
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
  it('should go to \'Locations > States\' page', async function () {
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 23]
```diff
@@ -0,0 +1,374 @@
...
+    await zonesPage.goToSubTabStates(page);
+
+    const pageTitle = await statesPage.getPageTitle(page);
+    expect(pageTitle).to.contains(statesPage.pageTitle);
+
+    const jsErrors = utilsPlaywright.getJsErrors();
+    expect(jsErrors.length).to.equals(0);
+  });
+
+  it('should go to add new state page', async function () {
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
  it('should go to \'Locations > States > New State\' page', async function () {
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 24]
```diff
@@ -0,0 +1,374 @@
...
+    await statesPage.goToAddNewStatePage(page);
+
+    const pageTitle = await addStatePage.getPageTitle(page);
+    expect(pageTitle).to.contains(addStatePage.pageTitleCreate);
+
+    const jsErrors = utilsPlaywright.getJsErrors();
+    expect(jsErrors.length).to.equals(0);
+  });
+
+  it('should go to edit States page', async function () {
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
  it('should go to \'Locations > States > Edit State\' page', async function () {
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 25]
```diff
@@ -0,0 +1,374 @@
...
+    );
+
+    const pageTitle = await taxesPage.getPageTitle(page);
+    expect(pageTitle).to.contains(taxesPage.pageTitle);
+
+    const jsErrors = utilsPlaywright.getJsErrors();
+    expect(jsErrors.length).to.equals(0);
+  });
+
+  it('should go to add new tax page', async function () {
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
  it('should go to \'International > Taxes > Add Tax\' page', async function () {
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 26]
```diff
@@ -0,0 +1,374 @@
...
+    await taxesPage.goToAddNewTaxPage(page);
+
+    const pageTitle = await addTaxPage.getPageTitle(page);
+    expect(pageTitle).to.contains(addTaxPage.pageTitleCreate);
+
+    const jsErrors = utilsPlaywright.getJsErrors();
+    expect(jsErrors.length).to.equals(0);
+  });
+
+  it('should go to edit Taxes page', async function () {
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
  it('should go to \'International > Taxes > Edit Tax\' page', async function () {
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 27]
```diff
@@ -0,0 +1,374 @@
...
+    await taxesPage.goToEditTaxPage(page, 1);
+
+    const pageTitle = await addTaxPage.getPageTitle(page);
+    expect(pageTitle).to.contains(addTaxPage.pageTitleEdit);
+
+    const jsErrors = utilsPlaywright.getJsErrors();
+    expect(jsErrors.length).to.equals(0);
+  });
+
+  it('should go to \'Tax Rules\' page', async function () {
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
  it('should go to \'International > TaxRules\' page', async function () {
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 28]
```diff
@@ -0,0 +1,374 @@
...
+    await taxesPage.goToTaxRulesPage(page);
+
+    const pageTitle = await taxRulesPage.getPageTitle(page);
+    expect(pageTitle).to.contains(taxRulesPage.pageTitle);
+
+    const jsErrors = utilsPlaywright.getJsErrors();
+    expect(jsErrors.length).to.equals(0);
+  });
+
+  it('should go to Add new tax rules group page', async function () {
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
  it('should go to \'International > TaxRules > Add Tax Rule Group\' page', async function () {
```

File: tests/UI/campaigns/audit/BO/11_international.ts [hunk 29]
```diff
@@ -0,0 +1,374 @@
...
+    await taxRulesPage.goToAddNewTaxRulesGroupPage(page);
+
+    const pageTitle = await addTaxRulesPage.getPageTitle(page);
+    expect(pageTitle).to.contains(addTaxRulesPage.pageTitleCreate);
+
+    const jsErrors = utilsPlaywright.getJsErrors();
+    expect(jsErrors.length).to.equals(0);
+  });
+
+  it('should go to edit Tax Rules page', async function () {
```
Comment (Progi1984) [inline tests/UI/campaigns/audit/BO/11_international.ts]:
```suggestion
  it('should go to \'International > TaxRules > Edit Tax Rule Group\' page', async function () {
```

**Review by Progi1984 — COMMENTED** [review]
@florine2623 Just changed some titles of test

**Review by Progi1984 — APPROVED** [review]

**Review by nesrineabdmouleh — APPROVED** [review]

**Pull request merged**